api/swagger.yaml

```diff
@@ -926,6 +926,13 @@ definitions:
               with option 'Capabilities'.
             items:
               type: "string"
+          Capabilities:
```

Comment:
Yet have to have a full look at the changes, but it looks like this PR is adding the `Capabilities` field again, which I think was removed in https://github.com/moby/moby/pull/41249.

What I think needs to happen is that the daemon configuration takes the place of oci.DefaultCapabilities();  https://github.com/moby/moby/blob/10866714412aea1bb587d1ad14b2ce1ba4cf4308/oci/caps/defaults.go#L4

Any `CapAdd` / `CapDrop` set on the container (through `--cap-add` / `--cap-drop`) should then diff relative to those defaults.

One question remains there, and that's how the custom capabilities will be visible to the user (because, like the current default capabilities), they're not visible (only `CapAdd` and `CapDrop` are, but there's no `docker inspect` that shows the effective capabilities.

An alternative could be to fill in `CapAdd` / `CapDrop` to achieve the same, something like

- `CapDrop: ALL` and `CapAdd: ["list of custom default caps"]` if no custom capabilities are set on the container
- Otherwise, take the configured default capabilities as starting point, and 

Looks like this PR still needs a rebase to get rid of the empty/unrelated commits that are in here

Reply:
Forgive my lack of context as I have not been following the docker development in detail over the past year. 
I believe you have three issues here: The name of the flag, the behavior and implementation. 

I am flexible on the name of the flag. If DefaultCapabilities is more clear I am fine with using that.

The current behavior is to set the default caps to the list provided or if none is provided take the default you mentioned. If a container does a cap add, that will be in addition to the new default set. Same with drop.

For the implementation, I went with create.go as we are already configuring HostConfig in that section. If I go with changing the defaults.go I will need add a reference to daemon.configStore in the call. This is fine as long as you guys are ok with this change.

Lastly I am not clear on what you are looking for with this comment:
"Looks like this PR still needs a rebase to get rid of the empty/unrelated commits that are in here"
Can you be more specific?

Reply:
You probably want check what was done on #41249 
What comes to rebase request "commits" tab says that there is 689 commits invluded to this PR as there should just one.

Reply:
Would it be easier to just open a new PR? This is a small change on an old branch so I would rather not deal with the noise that comes with a rebase. (When I did the rebase, it changes a bunch of unrelated files as this started about a year ago.)

Reply:
We prefer to stay on this PR. You should get rid of noise by doing rebase with squash for your commits (=> combine those changes to one commit).

If anything else does not work you can:
- create local backup branch (new one based on your current branch),
- switch back to this branch
- do hard reset for your branch to time before your first commit
- merge upstream master to your branch
- cherry-pick your changes from backup repo
- force push updated branch back to github

Reply:
Will do thanks.

Reply:
Well I got rid of the extra commits from the merge on he 21st but it looks like it took most of my changes. :) I'll fiddle with git to see if I can get them back.

Reply:
There is at least those messages like "burnMyDread force-pushed the burnMyDread:39149-Default-Capabilites branch from d093fef to ad97144 6 days ago" in here where commit ids works as links where you can go back and browse old versions.

Reply:
I had a local copy out of paranoia so I did it the old fashioned way. This is now sorted from a git PoV. Hoping to hear back on the content.

Reply:
Ok so now this is sorted from the git side and the tests are passing I wanted to re-add he context on why we want this change. I work for a InfoSec group and we follow the CIS benchmark guidance. We would like to have a way to drop CAP_NET_RAW and CAP_DAC_OVERRIDE by default rather than making each user drop all then add for each container. Or more specifically we want to be able to configure docker to be more secure by default.

Reply:
@theJeztah Can you comment on my questions from last week?